experimental/tinystr_neo/src/macros.rs

```diff
@@ -0,0 +1,27 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[macro_export]
+macro_rules! tinystr {
```

Comment:
Thought: We maybe don't need the macro any more. People can just use the const function constructors directly. Fewer imports and friendlier to the compiler.

Reply:
So as mentioned in https://github.com/unicode-org/icu4x/pull/1511#discussion_r785461338 , I think it's important to give users a convenient way to call this in a const context, given that it's rare and unusual for Rust programmers to `const` locals wherever possible. Unlike that PR, there will be no exported consts for people to use this, so I'd prefer to keep the macro around.

MBEs (non proc macros, like this one) are pretty friendly to the compiler; and this is going to be even smaller once I rebase over the rust update. I don't find that argument that compelling.

Reply:
Yeah, `macro_rules` is definitely an improvement over a proc macro.  Okay, I'm convinced.